src/CmsBundle/Security/AccessVoter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Opifer\CmsBundle\Security;
+
+use Opifer\CmsBundle\Entity\User;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Security;
+
+class AccessVoter extends Voter
+{
+    private $security;
+    private $container;
+
+    public function __construct(Security $security, ContainerInterface $container)
+    {
+        $this->security = $security;
+        $this->container= $container;
+    }
+
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+
+        if (!$user instanceof User) {
+            // the user must be logged in; if not, deny access
+            return false;
+        }
+
+        foreach($user->getRoles() as $role){
+            $rights = $this->container->getParameter('opifer_cms.permissions.'.strtolower($role));
+            if(in_array($attribute, $rights)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function supports($attribute, $subject)
+    {
+        if(null === $this->security->getUser()) {
+            return false;
+        }
+        foreach($this->security->getUser()->getRoles() as $role) {
+            if(null === $this->container->getParameter('opifer_cms.permissions.'.strtolower($role))) {
+                return false;
```

Comment:
I'm not sure what you're trying to do here... 

The `supports` method should check if this Voter should be used or not. So, you'll have to either check the `opifer_cms` config to see if the `attribute` is provided to any of the roles (not any of the roles of the current user). Or (a better solution IMO), you'll have to define an array somewhere with all possible actions that can be checked against. This could be a class with a predefined list, that you can extend per project.